pkg/wait/wait.go

```diff
@@ -48,7 +48,7 @@ func waitForApp(ctx context.Context, c client.Client, app *apiv1.App) (*apiv1.Ap
 	}
 	w := objwatcher.New[*apiv1.App](wc)
 	return w.ByObject(ctx, app, func(app *apiv1.App) (bool, error) {
-		if app.Status.Ready {
+		if app.Status.Ready && app.Generation == app.Status.ObservedGeneration {
```

Comment:
This was the key to get the whole thing working. The rest of the changes are to make it look correct in the terminal (i.e. not printing `OK` when we are still waiting and making sure the volume status is correctly reported)